frontend/src/components/utils/CountryFlag.tsx

```diff
@@ -20,22 +25,53 @@ const DefaultCountryFlag = () => (
   </div>
 );
 
-const CountryFlag = (props: Props) => (
-  <>
-    {props.countryCode ? (
-      <ReactCountryFlag
-        {...props}
-        svg
-        countryCode={props.countryCode}
-        style={{
-          width: "22px",
-          lineHeight: "14px",
-        }}
-      />
-    ) : (
-      <DefaultCountryFlag />
-    )}
-  </>
-);
+const CountryFlag = (props: Props) => {
+  // sometimes in 'country' can be actually countryCode
+  const mayBeCountryCode: string | undefined =
+    props.country &&
+    typeof props.country !== undefined &&
+    countries[props.country.toUpperCase()]
+      ? props.country.toUpperCase()
+      : undefined;
+  const countryCode: string | undefined =
+    props.countryCode && typeof props.countryCode !== undefined
+      ? props.countryCode.toUpperCase()
+      : mayBeCountryCode
+      ? mayBeCountryCode
+      : undefined;
```

Comment:
I find this mix of conditions unreadable

```suggestion
  let countryCode: string | undefined = props.countryCode?.toUpperCase();
  if (!(countryCode in countries)) {
    const countryAsCountryCode = props.country?.toUpperCase();
    if (countryAsCountryCode in countries) {
      countryCode = countryAsCountryCode;
    }
  }
```